echemdb/data/cv/descriptor.py

```diff
@@ -0,0 +1,122 @@
+class GenericDescriptor:
+    r"""
+    Wrapper for a data package's descriptor to make searching in metadata easier.
+
+    EXAMPLES::
+
+        >>> GenericDescriptor({'a': 0})
+        {'a': 0}
+
+    """
+    def __init__(self, descriptor):
+        self._descriptor = descriptor
+
+    def __dir__(self):
+        r"""
+        Return the attributes of this descriptor.
+
+        Implemented to allow tab-completion in a package's descriptor.
+
+        EXAMPLES::
+
+            >>> descriptor = GenericDescriptor({'a': 0})
+            >>> 'a' in dir(descriptor)
+            True
+
+        """
+        return list(key.replace(' ', '_') for key in self._descriptor.keys()) + object.__dir__(self)
+
+    def __getattr__(self, name):
+        r"""
+        Return the attribute `name` of the descriptor.
+
+        EXAMPLES::
+
+            >>> descriptor = GenericDescriptor({'a': 0})
+            >>> descriptor.a
+            0
+
+        """
+        name = name.replace('_', ' ')
+        if name in self._descriptor:
+            value = self._descriptor[name]
+            return Descriptor(value) if isinstance(value, dict) else value
+
+        raise AttributeError(f"Descriptor has no entry {name}. Did you mean one of {list(self._descriptor.keys())}?")
+
+    def __getitem__(self, name):
+        r"""
+        Return the attribute `name` of the descriptor.
+
+        EXAMPLES::
+
+            >>> descriptor = GenericDescriptor({'a': 0})
+            >>> descriptor["a"]
+            0
+
+        """
+        if name in self._descriptor:
+            value = self._descriptor[name]
+            return Descriptor(value) if isinstance(value, dict) else value
+
+        raise KeyError(f"Descriptor has no entry {name}. Did you mean one of {list(self._descriptor.keys())}?")
+
+    def __repr__(self):
+        r"""
+        Return a printable representation of this descriptor.
+
+        EXAMPLES::
+
+            >>> GenericDescriptor({})
+            {}
+
+        """
+        return repr(self._descriptor)
+
+    @property
+    def yaml(self):
+        r'''Return a printable representation of this descriptor in yaml format.
+
+        EXAMPLES::
+
+            >>> descriptor = GenericDescriptor({'a': 0})
+            >>> descriptor.yaml
+            'a: 0\n'
+
+        '''
+        import yaml
+        return yaml.dump(self._descriptor)
+
+class UnitValueDescriptor(GenericDescriptor):
```

Comment:
This class is new. A descriptor wrapper for descriptors that have `value` and `unit`.